src/Simplify.elm

```diff
@@ -6266,25 +6270,84 @@ arrayGetChecks checkInfo =
                 Nothing ->
                     Nothing
         , \() ->
-            case Evaluate.getInt checkInfo checkInfo.firstArg of
-                Just 0 ->
-                    Nothing
-
-                Just intValue ->
-                    callWithNonPositiveIntCanBeReplacedByCheck
-                        { int = intValue
-                        , intDescription = "index"
-                        , replacement = maybeWithJustAsWrap.empty.asString
-                        , lastArg = secondArg checkInfo
-                        }
-                        checkInfo
-
-                _ ->
-                    Nothing
+            Evaluate.getInt checkInfo checkInfo.firstArg
+                |> Maybe.andThen (indexAccessChecks arrayCollection checkInfo)
         ]
         ()
 
 
+indexAccessChecks : IndexableProperties otherProperties -> CheckInfo -> Int -> Maybe (Error {})
+indexAccessChecks collection checkInfo n =
+    if n < 0 then
+        Just
+            (alwaysResultsInUnparenthesizedConstantError (qualifiedToString checkInfo.fn ++ " with negative index")
+                { replacement = maybeWithJustAsWrap.empty.asString, lastArg = secondArg checkInfo }
+                checkInfo
+            )
+
+    else
+        case secondArg checkInfo of
+            Just arg ->
+                case collection.literalElements checkInfo.lookupTable arg of
+                    Just literalElements ->
+                        case Array.get n (Array.fromList literalElements) of
+                            Just element ->
+                                let
+                                    just : String
+                                    just =
+                                        qualifiedToString (qualify ( [ "Maybe" ], "Just" ) checkInfo)
+                                in
+                                Just
+                                    (Rule.errorWithFix
+                                        { message = "The element returned by " ++ qualifiedToString checkInfo.fn ++ " is known"
+                                        , details = [ "You can replace this call by Just the targeted element." ]
+                                        }
+                                        checkInfo.fnRange
+                                        (keepOnlyFix { parentRange = checkInfo.parentRange, keep = Node.range element }
```

Comment:
why not `replaceBySubExpression` and remove the paren insertion?

Reply:
Yeah, that was much nicer. At the moment, I find it hard to know which checks, fixes and other helpers to reach for. I try to look at how similar functions do things and I try to find things by name, but I often still miss things. I originally copied this from somewhere (and then added the parens insertion).

Reply:
Would be nice to have a local `Fix.Extra` with all our nice fix helpers maybe.